packages/upload-client/src/index.js

```diff
@@ -63,13 +67,16 @@ export async function uploadFile(conf, file, options = {}) {
  * has the capability to perform the action.
  *
  * The issuer needs the `store/add` and `upload/add` delegated capability.
- * @param {import('./types.js').FileLike[]} files File data.
+ * @param {Iterable<FileLike> & { sorted?: boolean }} files  Files that should be in the directory.
```

Comment:
I'm not 100% sure it's desirable to allow streaming the listing. If the files are not verified to be ordered correctly at the start, then we run the risk of uploading 1 or more shards before failing.

...this is ignoring the fact that `UnixFS.createDirectoryEncoderStream(files)` iterates over the files immediately.